### ISSUE TITLE
code can not run

### ISSUE THREAD

**Issue by NoTargetFish** [mperezcarrasco/PyTorch-BiGAN#1]
your code can not run with an error: one of the variables needed for gradient computation has been modified by an inplace operation, you need to forward and backward again when update generator, and I change it to this:
#Cleaning gradients.
                optimizer_d.zero_grad()

                #Generator:
                z_fake = Variable(torch.randn((x.size(0), self.args.latent_dim, 1, 1)).to(self.device),
                                  requires_grad=False)
                x_fake = self.G(z_fake)

                #Encoder:
                x_true = x.float().to(self.device)
                z_true = self.E(x_true)

                #Discriminator
                out_true = self.D(x_true + noise1, z_true)
                out_fake = self.D(x_fake + noise2, z_fake)

                #Losses
                if self.args.wasserstein:
                    loss_d = - torch.mean(out_true) + torch.mean(out_fake)
                else:
                    loss_d = criterion(out_true, y_true) + criterion(out_fake, y_fake)

                #Computing gradients and backpropagate.
                loss_d.backward()
                optimizer_d.step()
                
                #Cleaning gradients.
                optimizer_ge.zero_grad()

                #Generator:
                z_fake = Variable(torch.randn((x.size(0), self.args.latent_dim, 1, 1)).to(self.device),
                                  requires_grad=False)
                x_fake = self.G(z_fake)

                #Encoder:
                x_true = x.float().to(self.device)
                z_true = self.E(x_true)

                #Discriminator
                out_true = self.D(x_true + noise1, z_true)
                out_fake = self.D(x_fake + noise2, z_fake)
                
                #Losses
                if self.args.wasserstein:
                    loss_ge = - torch.mean(out_fake) + torch.mean(out_true)
                else:
                    loss_ge = criterion(out_fake, y_true) + criterion(out_true, y_fake)
                
                loss_ge.backward()
                optimizer_ge.step()
then it can run.
